### PR TITLE
Add config for PunctuationSpacing

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Sniff/PunctuationSpacingSniff.php [hunk 1]
```diff
@@ -35,20 +72,16 @@ protected function shouldHaveSpaceAfter(int $tokenPosition, array $tokens): ?int
...
 
-        if ($this->isTokenMatching($token, Token::PUNCTUATION_TYPE, self::ONE_SPACE_AFTER)) {
-            // We cannot add one space after, if the next token need zero space before: `[1,2,3,]`.
-            $nextPosition = $this->findNext(Token::WHITESPACE_TOKENS, $tokens, $tokenPosition + 1, true);
-            if (false !== $nextPosition && null !== $this->shouldHaveSpaceBefore($nextPosition, $tokens)) {
-                return null;
-            }
-
-            return 1;
+        // We cannot change spaces after one token, if the next one have a constraint: `[1,2,3,]`.
```
Comment (JoolsMcFly) [inline src/Sniff/PunctuationSpacingSniff.php:79]: `// We cannot change spaces after a token if the next one has a constraint: `[1,2,3,]`.`

Nitpicking, I know :)

File: src/Sniff/PunctuationSpacingSniff.php [hunk 2]
```diff
@@ -11,21 +11,58 @@
...
+        array $spaceAfterOverride = []
+    ) {
+        $this->spaceBeforeConfig = array_merge(self::DEFAULT_CONFIG_BEFORE, $spaceBeforeOverride);
+        $this->spaceAfterConfig = array_merge(self::DEFAULT_CONFIG_AFTER, $spaceAfterOverride);
+    }
 
     /**
      * @param list<Token> $tokens
      */
     protected function shouldHaveSpaceBefore(int $tokenPosition, array $tokens): ?int
```
Comment (JoolsMcFly) [inline src/Sniff/PunctuationSpacingSniff.php:58]: Given this method's name I would expect a boolean as return value.

Comment (VincentLanglet) [reply src/Sniff/PunctuationSpacingSniff.php:58]: Indeed

Comment (VincentLanglet) [reply src/Sniff/PunctuationSpacingSniff.php]: WDYT about `getExpectedSpacesAfter` and `getExpectedSpaceBefore` @JoolsMcFly ? Something else ?

Comment (JoolsMcFly) [reply src/Sniff/PunctuationSpacingSniff.php]: Or simply `getSpacesAfter` and `getSpacesBefore`?

**Comment by VincentLanglet** [comment]
Closing due to lack of interest

**Pull request closed without merging**